app/models/cherrypick_request.rb

```diff
@@ -5,74 +5,35 @@
 # Copyright (C) 2007-2011,2013,2014,2015 Genome Research Ltd.
 
 # This class is due to replace CherrypickForPulldownRequest
-class CherrypickRequest < TransferRequest
-  redefine_aasm column: :state, whiny_persistence: true do
-    # The statemachine for transfer requests is more promiscuous than normal requests, as well
-    # as being more concise as it has less states.
-    state :pending, initial: true
-    state :started
-    state :failed, enter: :on_failed
-    state :passed
-    state :cancelled, enter: :on_cancelled
-    state :hold
+class CherrypickRequest < CustomerRequest
+  after_create :build_stock_well_links, :transfer_aliquots
 
-    event :hold do
-      transitions to: :hold, from: [:pending]
-    end
-
-    # State Machine events
-    event :start do
-      transitions to: :started, from: [:pending, :hold]
-    end
-
-    event :pass do
-      transitions to: :passed, from: [:pending, :started, :failed]
-    end
-
-    event :fail do
-      transitions to: :failed, from: [:pending, :started, :passed]
-    end
-
-    event :cancel do
-      transitions to: :cancelled, from: [:started, :passed]
-    end
-
-    event :cancel_before_started do
-      transitions to: :cancelled, from: [:pending, :hold]
-    end
-
-    event :submission_cancelled do
-      transitions to: :cancelled, from: [:pending, :cancelled]
-    end
+  def on_started
+    # Aliquots are transferred on creation by transfer requests.
+    # This isn't ideal, but makes the transition easier without
+    # slowing actual picks down.
+  end
 
-    event :detach do
-      transitions to: :pending, from: [:pending, :cancelled]
-    end
+  def on_passed
+    target_asset.transfer_requests_as_target.where(submission_id: submission_id).find_each(&:pass!)
   end
 
-  def on_failed
-    # Do nothing
+  def reduce_source_volume
+    return unless asset.get_current_volume
+    subtracted_volume = target_asset.get_picked_volume
+    new_volume = asset.get_current_volume - subtracted_volume
+    asset.set_current_volume(new_volume)
   end
 
-  alias_method :on_cancelled, :on_failed
+  private
 
-  def perform_transfer_of_contents
-    transfer_aliquots # Ensures we set the study/project
+  # The transfer requests handle the actual transfer
```

Comment:
So the creation of transfer requests may seem a little redundant, but is a limitation in the way state is tracked in SS that we probably need to address. Essentially well state exists on the transfer requests into the well.This allows individual well failure, and I think its position on request, rather than the well (different wells on the same plate can have different states) itself was an attempt to consolidate state machines in one location, and to have everything be request driven. When Generic Lims was implimented there was a need to fail wells on the stock plate, before any work had been completed. At this point, cherypick requests became a subclass of transfer request. This isn't great, and we probably need to completely rethink how sll this gets modeled.